File: app/src/main/java/com/example/University_blind_dating/ui/hearts/HeartsFragment.java
package com.example.University_blind_dating.ui.hearts;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

import com.example.University_blind_dating.R;

public class HeartsFragment extends Fragment {

    private HeartsViewModel heartsViewModel;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        heartsViewModel = ViewModelProviders.of(this).get(HeartsViewModel.class);
        View root = inflater.inflate(R.layout.fragment_hearts, container, false);

        return root;
    }
    /*------------------Actionbar of Home Fragment-------------------*/
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        setHasOptionsMenu(true);
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        inflater.inflate(R.menu.actionbar_actions, menu);
        // hide item (settings)
        menu.findItem(R.id.user_settings).setVisible(false);
        super.onCreateOptionsMenu(menu, inflater);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        // Actionbar item click event
        int id = item.getItemId();
        if (id == R.id.home_search){
            Toast.makeText(getActivity(), "Search", Toast.LENGTH_SHORT).show();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/example/myapplication/ui/hearts/HeartsFragment.kt
package com.example.myapplication.ui.hearts

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.example.myapplication.R

class HeartsFragment : Fragment() {

    private lateinit var heartsViewModel: HeartsViewModel

    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {
        heartsViewModel =
                ViewModelProviders.of(this).get(HeartsViewModel::class.java)
        val root = inflater.inflate(R.layout.fragment_hearts, container, false)
        val textView: TextView = root.findViewById(R.id.text_hearts)
        heartsViewModel.text.observe(viewLifecycleOwner, Observer {
            textView.text = it
        })
        return root
    }
}